crates/rpc/rpc-eth-api/src/api/servers/helpers/call.rs

```diff
@@ -0,0 +1,29 @@
+//! Contains RPC handler implementations specific to endpoints that call/execute within evm.
+
+use reth_evm::ConfigureEvm;
+
+use crate::{
+    servers::{Call, EthCall, LoadPendingBlock, LoadState, SpawnBlocking},
+    EthApi,
+};
+
+impl<Provider, Pool, Network, EvmConfig> EthCall for EthApi<Provider, Pool, Network, EvmConfig> where
+    Self: Call + LoadPendingBlock
+{
+}
+
+impl<Provider, Pool, Network, EvmConfig> Call for EthApi<Provider, Pool, Network, EvmConfig>
+where
+    Self: LoadState + SpawnBlocking,
+    EvmConfig: ConfigureEvm,
+{
+    #[inline]
+    fn call_gas_limit(&self) -> u64 {
+        self.inner.gas_cap()
+    }
+
+    #[inline]
+    fn evm_config(&self) -> &impl ConfigureEvm {
```

Comment:
it doesn't work because `ConfigureEvm` is bound by `'static` lifetime, and `impl ConfigureEvm` can then resolve to a type with `'static` lifetime. so removing the reference on the opaque type would require cloning the config on each call to `evm_config`, which is an unnecessary perf downgrade.